rundeckapp/src/test/groovy/rundeck/services/UserServiceSpec.groovy

```diff
@@ -443,4 +443,16 @@ class UserServiceSpec extends Specification implements ServiceUnitTest<UserServi
         new Date() | null         | 3
         new Date() | baseDate     | 0
     }
+
+    def "validateUserNameWithSingleQuotes"() {
```

Comment:
add a test for a non valid username also

Reply:
added!